st2actions/st2actions/handlers/mistral.py

```diff
@@ -49,30 +61,35 @@ def callback(url, context, status, result):
             return
 
         try:
-            method = 'PUT'
-            headers = {'content-type': 'application/json'}
-
             if isinstance(result, basestring) and len(result) > 0 and result[0] in ['{', '[']:
                 value = ast.literal_eval(result)
                 if type(value) in [dict, list]:
                     result = value
 
+            action_execution_id = get_action_execution_id_from_url(url)
             output = json.dumps(result) if type(result) in [dict, list] else str(result)
             data = {'state': STATUS_MAP[status], 'output': output}
 
+            client = mistral.client(
+                mistral_url=cfg.CONF.mistral.v2_base_url,
+                username=cfg.CONF.mistral.keystone_username,
+                api_key=cfg.CONF.mistral.keystone_password,
+                project_name=cfg.CONF.mistral.keystone_project_name,
+                auth_url=cfg.CONF.mistral.keystone_auth_url)
+
+            manager = action_executions.ActionExecutionManager(client)
+
             for i in range(cfg.CONF.mistral.max_attempts):
                 try:
                     LOG.info('Sending callback to %s with data %s.', url, data)
-                    response = requests.request(method, url, data=json.dumps(data), headers=headers)
-                    if response.status_code == 200:
-                        break
+                    manager.update(action_execution_id, **data)
+                    break
                 except requests.exceptions.ConnectionError as conn_exc:
-                    LOG.exception(conn_exc)
-                    if i < cfg.CONF.mistral.max_attempts:
+                    if i == cfg.CONF.mistral.max_attempts - 1:
+                        raise conn_exc
+                    else:
+                        LOG.exception(conn_exc)
```

Comment:
I know you didn't change this in this PR, but it seems it would be better to log this under warn or similar, since we haven't reached max attempts yet and this exception is not fatal yet.